src/v/cluster/types.h

```diff
@@ -904,6 +904,14 @@ struct config_update_reply {
     cluster::config_version latest_version{config_version_unset};
 };
 
+struct hello_request final {
+    model::node_id peer;
```

Comment:
Could we add the process startup time here?  That would enable the receive to proactively clean up things from before the restart (e.g. tear down TCP connections we can infer to be dead).

(maybe just a placeholder for encoding purposes if you're short on time)

Reply:
Thinking about this further: is there a benefit to a sender tagging the request with its local wall clock time vs the receiver using its own wall clock time when it handles the hello request?

Reply:
I think so, because of the delta between process startup and sending the request.  The receiver should be able to assume that anything from before time T originates from the previous incarnation of the node.

Considering the case I have in mind for cleaning up TCP connections, by the time we receive the hello message it's too late: the node already established a connection with us to send the hello, so I can't say "connections from before time T are stale".

Reply:
I was struggling with "connections from before time T are stale" because if the node hadn't started the connection couldn't exist. But I wasn't considering things like transports in a backoff state or any other state that we might later invent. Ok this makes sense. Thanks!